flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/sink/shuffle/TestShuffleOperator.java

```diff
@@ -0,0 +1,166 @@
+        1,
+        new TestTaskStateManager());
+  }
+
+  @Before
+  public void before() throws Exception {
+    MockOperatorEventGateway mockGateway = new MockOperatorEventGateway();
+    KeySelector<String, String> keySelector =
+        new KeySelector<String, String>() {
+          private static final long serialVersionUID = 7662520075515707428L;
+
+          @Override
+          public String getKey(String value) {
+            return value;
+          }
+        };
+    DataStatisticsFactory<String> dataStatisticsFactory = new DataStatisticsFactory<>();
+
+    this.operator = new ShuffleOperator<>(keySelector, mockGateway, dataStatisticsFactory);
+    Environment env = getTestingEnvironment();
+    this.operator.setup(
+        new OneInputStreamTask<String, String>(env),
+        new MockStreamConfig(new Configuration(), 1),
+        new MockOutput<>(Lists.newArrayList()));
+  }
+
+  @After
+  public void clean() throws Exception {
+    operator.close();
+  }
+
+  @Test
+  public void testInitializeState() throws Exception {
+    StateInitializationContext stateContext = getStateContext();
+    operator.initializeState(stateContext);
+
+    assertNotNull(
+        stateContext
+            .getOperatorStateStore()
+            .getListState(operator.generateGlobalDataDistributionWeightDescriptor()));
+  }
+
+  @Test
+  public void testProcessElement() throws Exception {
+    StateInitializationContext stateContext = getStateContext();
+    operator.initializeState(stateContext);
+    operator.processElement(new StreamRecord<>("a"));
+    operator.processElement(new StreamRecord<>("a"));
+    operator.processElement(new StreamRecord<>("b"));
+    assertTrue(operator.localDataStatistics() instanceof MapDataStatistics);
+    MapDataStatistics<String> mapDataStatistics =
+        (MapDataStatistics<String>) operator.localDataStatistics();
+    assertTrue(mapDataStatistics.dataStatistics().containsKey("a"));
+    assertTrue(mapDataStatistics.dataStatistics().containsKey("b"));
+    assertEquals(2L, (long) mapDataStatistics.dataStatistics().get("a"));
+    assertEquals(1L, (long) mapDataStatistics.dataStatistics().get("b"));
+  }
+
+  @Test
+  public void testOperatorOutput() throws Exception {
+    try (OneInputStreamOperatorTestHarness<String, ShuffleRecordWrapper<String, String>>
+        testHarness = createHarness(this.operator)) {
+      testHarness.processElement(new StreamRecord<>("a"));
+      testHarness.processElement(new StreamRecord<>("b"));
+      testHarness.processElement(new StreamRecord<>("b"));
+
+      List<String> recordsOutput =
+          testHarness.extractOutputValues().stream()
+              .filter(ShuffleRecordWrapper::hasRecord)
+              .map(ShuffleRecordWrapper::record)
+              .collect(Collectors.toList());
+      assertThat(recordsOutput)
+          .containsExactlyInAnyOrderElementsOf(ImmutableList.of("a", "b", "b"));
+    }
+  }
+
+  // ---------------- helper methods -------------------------
```

Comment:
maybe remove this comment. doesn't seem to be the style that Iceberg code has

Reply:
will remove